### PR TITLE
Small fix to -Werror compilation issue

### PR DESCRIPTION
**Pull request by vsamy**
<sup> Created from VS Code using [CodeStream](https://codestream.com/?utm_source=cs&utm_medium=pr&utm_campaign=github*com)</sup>

### CODE REVIEW

**Review by gauthamm — APPROVED** [review]

File: sdk/src/rplidar_driver.cpp [hunk 1]
```diff
@@ -1792,10 +1792,10 @@ u_result RPlidarDriverImplCommon::grabScanData(rplidar_response_measurement_node
 
     switch (_dataEvt.wait(timeout))
     {
-    case rp::hal::Event::EVENT_TIMEOUT:
+    case static_cast<unsigned long>(rp::hal::Event::EVENT_TIMEOUT):
```
Comment (gauthamm) [inline sdk/src/rplidar_driver.cpp:1795]: enum
    {
        EVENT_OK = 1,
        EVENT_TIMEOUT = -1,
        EVENT_FAILED = 0,
    };
    It should not be unsigned

**Review by gauthamm — CHANGES_REQUESTED** [review]

File: sdk/src/rplidar_driver.cpp (hunk 1 above)
Comment (gauthamm) [reply sdk/src/rplidar_driver.cpp:1795]: Looking through the code, it seems like they messed up and used unsigned long eventhough the value is -1. I suggest changing the interface to a enum class and maybe even make a PR upstream

Comment (vsamy) [reply sdk/src/rplidar_driver.cpp:1795]: Oh, sorry, i updated too fast... I though the default value was -1.

Comment (vsamy) [reply sdk/src/rplidar_driver.cpp:1795]: They actually do the same for windows...
https://github.com/rapyuta-robotics/rplidar_ros/blob/rr-devel/sdk/src/hal/event.h#L108

Comment (gauthamm) [reply sdk/src/rplidar_driver.cpp:1795]: yup they should have used either an int8 or preferably use enum class. Not sure if this was already fixed upstream but can contribute the fix if it was not handled properly.

**Pull request closed without merging**